lib/SimpleSAML/Utils/Config/Metadata.php

```diff
@@ -300,6 +300,11 @@ public static function getDefaultEndpoint(array $endpoints, array $bindings = nu
      */
     public static function isHiddenFromDiscovery(array $metadata)
     {
+        assert(array_key_exists('EntityAttributes', $metadata));
+        assert(is_array($metadata['EntityAttributes']));
+        assert(array_key_exists(self::$ENTITY_CATEGORY, $metadata['EntityAttributes']));
+        assert(is_array($metadata['EntityAttributes'][self::$ENTITY_CATEGORY]));
+
```

Comment:
I think we shouldn't have assertions here. It's the same argument as before. Metadata might not have entity attributes at all, and in that case this method should just return false. Adding assertions here could lead to warnings in the logs in the best case, or errors if assertions are configured to fail in PHP, worst case scenario.

The code was designed to: _if there is such an entity category defined in the metadata, return true, and if not, return false_. This can be done in one single statement but can then produce warnings, which is why we're masking errors.

Reply:
Removed them